### ISSUE TITLE
Bad window look on macOS with light theme enabled when using dark theme in Zed

### ISSUE THREAD

**Issue by widersky** [zed-industries/zed#5049]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When I use a light theme in macOS, the dark theme in Zed looks not the best. On the screenshot below you can see an example - in the background is Slack, in the foreground Zed. Notice the lack of a dark 1px border around the Zed window.

![SCR-20230109-fn3](https://user-images.githubusercontent.com/45122321/211285829-ece897ea-2d84-421b-bb4f-5d679143118b.png)

### Environment

macOS 13.1, Zed 0.67.5

**Comment by michealp-coder** [comment]
I think I have the same issue too.

The window border on Zed (with a light them), is extremely "thick". 

It's a thick grey border (using the Atelier Sulphurpool Light, theme)

Where as any other macOS app (like Safari, Outlook, etc), the window has **no** border.

As such, it make the window drop shadow odd looking and make Zed look non-native.

**Comment by widersky** [comment]
@michealp-coder I'm not sure I understand correctly. My case is when macOS uses a light and Zed uses a dark theme. In this case, Zed has a kind of "blurred" edge:

![image](https://github.com/zed-industries/zed/assets/45122321/626ab1fd-d903-45b7-b28b-beee0da58c0a)

For comparison, the IntelliJ IDEA window:

![image](https://github.com/zed-industries/zed/assets/45122321/78f33f65-e954-4980-b8ec-2a2f9e416f7b)

**Comment by widersky** [comment]
FYI: VSCode is adding proper setting to fix this behavior in their [1.86 release](https://github.com/microsoft/vscode-docs/blob/vnext/release-notes/v1_86.md) ("New setting to enforce system color theme").

![image](https://github.com/zed-industries/zed/assets/45122321/70b163bb-b1be-42a7-8c6b-c69dbd0e9e2e)

**Comment by notpeter** [comment]
Circling back on this issue and wanted to clarify the actual request(s) here.

I think the matching native controls to whether the theme is dark/light is covered by this:
- https://github.com/zed-industries/zed/issues/13673

Other than that, we don't really use native controls (e.g. our context menus are native GPUI, not MacOS).

@widersky is this a request for the theme layer to add support for themes to have two border colors which are selected based on the MacOS dark mode / light mode setting? Or am I missing something.

**Comment by widersky** [comment]
@notpeter 

As you can see in the photo in the original post at the very top, the native window in macOS in dark mode has two single-pixel borders. This happens regardless of whether macOS has light or dark mode set - if the IDE has dark mode set, it behaves as if the entire macOS also had dark mode. Thanks to this, the window looks solid and does not have the impression that it is "blurred" at the edges. I only meant the appearance of the window - the appearance of context menus is not that important to me.

To sum up - if macOS has a light mode and the editor has a dark mode, Zed (when using one of the themes defined as dark) should look as if macOS had a dark mode set.

**Comment by notpeter** [comment]
> To sum up - if macOS has a light mode and the editor has a dark mode, Zed (when using one of the themes defined as dark) should look as if macOS had a dark mode set.

Apologies for being a design-blind (practically speaking), but what does "should look as if macOS had a dark mode set" mean and how would we implement this? Specifically what is different?

To my eye Zed looks similar to VSCode and Slack/JetBrains look like they have an ugly one pixel black border. But luckily I'm not a designer.
![image](https://github.com/user-attachments/assets/6ae4ae3c-ada3-459f-90e5-63d97f67d5bb)

**Comment by widersky** [comment]
![CleanShot 2024-09-03 at 17 56 22](https://github.com/user-attachments/assets/97198571-8203-40c0-bdfc-734d2ea65661)

There are 4 windows in the attached screenshot. 3 editors/IDEs (1) and a Finder fragment (2) indicating that macOS is in light mode. VSCode and PHPStorm with a dark theme display the correct border, while Zed does not.

![CleanShot 2024-09-03 at 17 59 53](https://github.com/user-attachments/assets/edbf4e98-faff-4529-a428-c87905c8119d)

Above, a similar arrangement of windows when macOS has dark mode enabled - in such a scenario, the Zed window looks fine.

**Comment by include** [comment]
+1 everything got "blur" :)
